osu.Game.Tests/Visual/Menus/TestSceneToolbar.cs

```diff
@@ -72,5 +74,26 @@ public void TestRulesetSwitchingShortcut(bool toolbarHidden)
                 AddUntilStep("ruleset switched", () => rulesetSelector.Current.Value.Equals(expected));
             }
         }
+
+        [TestCase(OverlayActivation.All)]
+        [TestCase(OverlayActivation.Disabled)]
+        public void TestRespectsOverlayActivation(OverlayActivation mode)
+        {
+            AddStep($"set activation mode to {mode}", () => toolbar.OverlayActivationMode.Value = mode);
+            AddStep("hide toolbar", () => toolbar.Hide());
+            AddStep("try to show toolbar", () => toolbar.Show());
+
+            if (mode == OverlayActivation.Disabled)
+                AddUntilStep("toolbar still hidden", () => toolbar.Visibility == Visibility.Hidden);
+            else
+                AddAssert("toolbar is visible", () => toolbar.Visibility == Visibility.Visible);
```

Comment:
could also be parameterised:

```suggestion
            AddUntilStep($"toolbar is {expectedVisibility}", () => toolbar.State.Value == expectedVisibility);
```

also why is one of these an assert and the other an until step?